server/hazard-service/src/main/java/edu/illinois/ncsa/incore/service/hazard/models/eq/utils/HazardCalc.java

```diff
@@ -291,8 +302,9 @@ public static SeismicHazardResult computeGroundMotionAtSite(Earthquake earthquak
 
             hazardValue = Math.exp(hazardValue);
 
-            // TODO check if site class dataset is defined for amplifying hazard
-            int siteClass = HazardUtil.getSiteClassAsInt(eqModel.getDefaultSiteClass());
+            // Get site class information if available, otherwise use the scenario default
+            int siteClass = site.getSiteClass() != null ? HazardUtil.getSiteClassAsInt(site.getSiteClass()) :
```

Comment:
what if invalid siteClass get passed in? getSiteClassAsInt will return -1; then it will be passed into 
```
 } else {
            int[] rows = getShortPeriodRows(hazardValue);
            if (rows[0] == rows[1]) {
                return shortPeriodFactors[rows[0]][siteClass];
            } else {
                return linearInterpolateAmplification(shortPeriodIntervals[rows[0]], shortPeriodFactors[rows[0]][siteClass],
                    shortPeriodIntervals[rows[1]], shortPeriodFactors[rows[1]][siteClass], hazardValue);
            }
        }
```

Will that be a problem?

Reply:
That's a good question. I can think of two areas issues can arise: 1) the site class map has an invalid entry (unexpected character) and we get -1 or a user creates a model EQ and sets an invalid site class as the default site class. The assumption is the user sets a valid site class inside the model EQ when they create it (the default) and the site class map, if it has an entry, it's valid. Those are the two points of potential failure. The easiest solution might be to change line 310 to be:
if(amplifyHazard && siteClass != -1) to prevent unsupported site classes. 

I could also add a check for -1 and print a log message. I don't think it's worth failing the entire post over this. Thoughts?

Reply:
Yeah could you add that condition to prevent unsupported site. And I will approve. Didn't test it though but the code looks good.

Reply:
@vnarah2 @longshuicy I made the change to prevent using -1. We could merge this as it is now or there is one other change I could make if we wanted. There is a case where site.getSiteClass() is non null, but invalid (empty or bad char), which would return a -1 and never hit the else statement where it gets the value from the model instead. I could split the code out to be:

// Get site class information if available, otherwise use the scenario default
int siteClass = site.getSiteClass() != null ? HazardUtil.getSiteClassAsInt(site.getSiteClass()) : -1;

// In case the site class is invalid or missing, use the model default site class, if available
if (siteClass == -1 && eqModel.getDefaultSiteClass() != null) {
       HazardUtil.getSiteClassAsInt(eqModel.getDefaultSiteClass());
}